src/connection.rs

```diff
@@ -1,313 +1,165 @@
-extern crate dbus;
-
-use std;
-use std::env;
+use std::rc::Rc;
+use std::cell::RefCell;
+use std::fmt;
 
 use enum_primitive::FromPrimitive;
 
-use device::{Device, DeviceType};
-use wifi::{AccessPoint, Security};
-use manager;
-use manager::NetworkManager;
-
-/// Get a list of Network Manager connections sorted by path.
-///
-/// # Examples
-///
-/// ```no_run
-/// use network_manager::connection;
-/// use network_manager::manager;
-/// let manager = manager::new();
-/// let connections = connection::list(&manager).unwrap();
-/// println!("{:?}", connections);
-/// ```
-pub fn list(manager: &NetworkManager) -> Result<Vec<Connection>, String> {
-    let paths = try!(manager.list_connections());
+use dbus_nm::DBusNetworkManager;
 
-    let mut connections = Vec::new();
+use wifi::Security;
+use device::{Device, get_active_connection_devices};
 
-    for path in &paths {
-        connections.push(try!(get_connection(&manager, path)))
-    }
 
-    connections.sort();
-
-    Ok(connections)
+#[derive(Clone)]
+pub struct Connection {
+    dbus_manager: Rc<RefCell<DBusNetworkManager>>,
```

Comment:
I was thinking about the same when I added the `RefCell` so that we allow dynamic changes, but we better go with construction only strategy and use only an `Rc`, which will simplify the code.